ground/gcs/src/plugins/uavtalk/uavtalk.cpp

```diff
@@ -123,23 +111,26 @@ UAVTalk::ComStats UAVTalk::getStats()
  */
 void UAVTalk::processInputStream()
 {
-    quint8 tmp;
-
-    if (io && io->isReadable()) {
-        while (io->bytesAvailable() > 0) {
-            io->read((char *)&tmp, 1);
-            processInputByte(tmp);
+    while (io && io->isReadable()) {
+        if (startOffset > (sizeof(rxBuffer) - MAX_PACKET_LENGTH)) {
+            /* If we're not sure there's room for a frame, shift things left in
```

Comment:
I disagree here-- I like vectors but the objective here is three things: A) copy almost all bytes once (which vector may or may not do for you depending on impl and semantics), B) don't call into the io device too much, and C) use a fixed, relatively small amount of buffer memory [but enough that copying can be limited for A and number of calls can be limited for B],

The cost here in skipping to a sync byte is very high, though-- I pretty much ignored it because I figured it should be rare.